examples/service-catalog/service-catalog.yaml

```diff
@@ -0,0 +1,370 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: service-catalog
+objects:
+
+- kind: ClusterRoleBinding
```

Comment:
@pmorie @deads2k @jwforres I think this template defines all the roles+bindings needed:

1) clusterrole/binding allowing anyone to call the template broker (needed until we have auth support for SC->broker flows)
2) clusterrole/binding allowing the SC SA to view all namespaces
3) a clusterrole/binding allowing the SC SA to update broker/instance/binding status
4) a role/binding allowing the SC SA to mange endpoints in its project (needed for etcd for now, should be removed eventually)
5) a clusterrole/binding allowing any authenticated user to view serviceclasses/instances/bindings (i'm thinking this should be a namespace scoped binding for users, not cluster?  but the SC SA will need a cluster scoped binding)
6) a clusterrole/binding allowing any authenticated user to create/update/delete serviceclasses/instances/bindings - same comment as (5)
7) a cluster role allowing CRUD on brokers/instances/bindings/serviceclasses - unused, not bound to anyone right now.

please review/critique/etc.  Assuming 5+6 should be namespaced for users, i'll need direction on how we make that happen (ie how to ensure users get the role bound when they create new projects).